### ISSUE TITLE
Improve UX of reopening assessment instances

### ISSUE THREAD

**Issue by nwalters512** [PrairieLearn/PrairieLearn#10727]
After an assessment instance has been closed (perhaps manually, perhaps via an Exam auto-closing after 6 hours of inactivity), reopening it is confusing:

<img width="261" alt="Screenshot 2024-09-30 at 11 02 59" src="https://github.com/user-attachments/assets/40d7f80b-dc95-4290-8b96-b896ca416b3f">

There are only options to set the total time limit or the remaining time limit. It's not clear what these mean, and it's not possible to reopen the assessment instance indefinitely.

I'm proposing making the following changes:

- Add an option to re-open without setting a time limit (that is, keep the instance open indefinitely).
- Add an option to set an absolute end date/time. The data model already supports this, as assessment instances track `date_limit`.
- For "total time" and "remaining time" add explanations + actual calculations to show the new end date.
- There aren't any options under the "minutes" dropdown. We should just display that as a fixed unit suffix.

It's worth noting that the "Change time limit for all instances" is perhaps a little closer to what we want, though I think we can still do better for the single-instance case:

<img width="592" alt="Screenshot 2024-09-30 at 11 07 59" src="https://github.com/user-attachments/assets/e104cbfc-73d3-42c2-b793-9c890e08b863">
<img width="587" alt="Screenshot 2024-09-30 at 11 13 01" src="https://github.com/user-attachments/assets/68a0ef4f-ec9e-4b13-87e2-e51359d2ca38">

**Comment by nwalters512** [comment]
> It would be great to unify these as much as possible. I was actually confused because I remembered the "remove time limit" option but I forgot that it was only available for all instances.

I agree in principle, but I actually the the kinds of operations that one would want to do in bulk don't overlap a whole lot with the kinds of things we'd want to do with individual instances. Or at the very least, the UI should look different (e.g. a checkbox like "Also reopen closed instances" doesn't make sense for an individual instances).

How about this as an idea:

<img width="293" alt="Screenshot 2024-09-30 at 11 50 32" src="https://github.com/user-attachments/assets/e9452e5e-c644-4718-9823-91be4b105a4b">

That is, we show "re-open without time limit" as the default option if the instance is closed. One can also select the other option, at which point we show the usual dropdowns (this mockup doesn't show any of the other proposed improvements):

<img width="318" alt="Screenshot 2024-09-30 at 11 50 46" src="https://github.com/user-attachments/assets/af7bb084-283e-43fc-a13f-fc83a2bb1a8c">

This admittedly drifts even further from the bulk-edit option, but I think it's much better for its particular purpose.

**Comment by mwest1066** [comment]
I'm on board with all of the above suggestions! (earlier I was confused between the single-instance and all-instance dialogs)
